src/google-calendar.ts

```diff
@@ -0,0 +1,250 @@
+// The options we are interested in are:
+//  - "EVENT" - this is the normal screen view of a calendar showing events. Within this screen
+//     pop up dialogs are shown with the details of events.
+//  - "EVENT_EDIT" - when you click the "edit" button on an event dialog in the EVENT screen, a
+//     full screen edit mode is displayed. This switches the view family to "EVENT_EDIT".
+export function getViewFamily(): string | undefined {
+  return document?.body?.dataset?.viewfamily;
+}
+
+export function isGoogleCalendar(): boolean {
+  return !!getViewFamily();
+}
+
+/*
+ * QUICK ADD SCREEN
+ */
+
+// The "quick add" screen is the inline event creation dialog,
+// invoked usually by clicking the "create +" button. We add a button
+// here, and get it to invoke the full-screen "edit" mode where the full functionaltiy is.
+function addButtonToQuickAdd(quickAddDialog: JQuery<HTMLElement>) {
+  // skip if our button is already added
+  if ($("#jitsi_button_quick_add").length) {
+    return;
+  }
+  const tabEvent = quickAddDialog.find("#tabEvent");
+  if (tabEvent.length) {
+    tabEvent.parent().append(
+      `
+    <content class="" role="tabpanel" id="jitsi_button_quick_add_content">
+      <div class="fy8IH poWrGb">
+        <div class="FkXdCf HyA7Fb">
+          <div class="DPvwYc QusFJf jitsi_quick_add_icon"/>
+        </div>
+      </div>
+      <div class="mH89We">
+        <div role="button"
+             class="uArJ5e UQuaGc Y5sE8d"
+             id="jitsi_button_quick_add">
+          <content class="CwaK9">
+            <span class="RveJvd jitsi_quick_add_text_size">
+              Add a Brave Talk meeting
+            </span>
+          </content>
+        </div>
+      </div>
+    </content>
+    `
+    );
+    const clickHandler = tabEvent.parent().find("#jitsi_button_quick_add");
+    clickHandler.on("click", () => {
+      // this is clicking the "more options" button on the quick add dialog,
+      // which causes the full screen event editor to appear
+      scheduleAutoCreateMeeting = true;
+      $('div[role="button"][jsname="rhPddf"]').trigger("click");
+    });
+  }
+}
+
+/*
+ * FULL SCREEN EVENT EDIT
+ */
+
+function isFullScreenEventButtonPresent(): boolean {
+  return document.getElementById("jitsi_button") !== null;
+}
+
+// the value currently entered into the "location" text box
+function getLocationString(): string {
+  return (
+    $("#xLocIn input[jsname=YPqjbf][role=combobox]").val()?.toString() ?? ""
+  );
+}
+
+async function setLocationString(newValue: string): Promise<void> {
+  const input = document.querySelector(
+    "#xLocIn input[jsname=YPqjbf][role=combobox]"
+  );
+
+  if (input instanceof HTMLElement) {
+    // inspired by
+    //  https://stackoverflow.com/questions/64094461/edit-descrption-or-location-of-google-calendar-event-with-chrome-extension
+
+    input.focus();
+
+    await new Promise((resolve) => setTimeout(resolve, 200));
+
+    document.execCommand("insertHTML", false, newValue);
```

Comment:
could you elaborate on why we are using execCommand? https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand says it is deprecated

also this calls insertHTML but looks like the newValue is just text - can it insert as text to avoid potential XSS vectors in newValue?

Reply:
Good suggestion, updated to `insertText` instead.

In terms of using execCommand, yes this is deprecated, but hours of trying alternative approaches did not find one that worked reliably: we need not just to update the content of the input box, but do it in such a way to causes the gcal javascript to process the entered input. Otherwise, the entry does not save when the meeting is saved.

As the MDN article notes:

> The Clipboard API can be used instead of execCommand in many cases, but execCommand is still sometimes useful. In particular, the Clipboard API doesn't replace the insertText command, which you can use to programmatically replace text at the cursor while preserving the undo buffer (edit history) in plain textarea and input elements.